nemo/core/classes/modelPT.py

```diff
@@ -772,7 +780,10 @@ def setup_optimization(
             self._optimizer = optimizer
 
         self._scheduler = prepare_lr_scheduler(
-            optimizer=self._optimizer, scheduler_config=scheduler_config, train_dataloader=self._train_dl
+            optimizer=self._optimizer,
+            scheduler_config=scheduler_config,
+            train_dataloader=self._train_dl,
+            ipl_config=self.cfg.get("ipl", None),
```

Comment:
Sorry but this cannot be added to the core. The core does not make any assumptions for it's domains. You'll need to think of a different way to implement this part.

Reply:
You can do it in a subclass of ASRModel, so that this is specific to that subclass

Reply:
Done